src/ota_proxy/ota_cache.py

```diff
@@ -111,24 +116,30 @@ def __init__(
         upper_proxy: str = "",
         enable_https: bool = False,
         external_cache_mnt_point: str | None = None,
+        remote_read_buffer_size: int = cfg.REMOTE_READ_BUFFER_SIZE,
+        max_concurrent_caching: int = cfg.MAX_CONCURRENT_CACHE_WRITES,
     ):
         """Init ota_cache instance with configurations."""
         logger.info(
             f"init ota_cache({cache_enabled=}, {init_cache=}, {upper_proxy=}, {enable_https=})"
         )
         self._closed = True
         self._shutdown_lock = asyncio.Lock()
+        self._caching_se = threading.Semaphore(max_concurrent_caching)
```

Comment:
Use `threading.Semphore` for limiting the number of cache writing tasks, as cache write now is executed within a cache write worker thread pool